front/src/modules/rollingStock/hooks/useFilterRollingStock.tsx

```diff
@@ -55,7 +55,7 @@ function rollingStockPassesSearchedStringFilter(
     return true;
   }
   function includesSearchedString(str: string | undefined) {
-    return str && str.toLowerCase().includes(filters.text);
+    return str && str.trim().toLowerCase().includes(filters.text.trim().toLowerCase());
```

Comment:
```suggestion
    return str?.trim().toLowerCase().includes(filters.text.trim().toLowerCase());
```

Reply:
I think this changes the type of the returned value: was `boolean` before, will be `boolean | undefined` with the suggested change. Maybe it doesn't matter too much though.

Reply:
`str && str.trim()` was already returning `boolean | undefined`
`&&` returns the left operand `str` if it is falsy, and the right operand otherwise

`undefined && (any expression)` evaluate as `undefined`

Reply:
Hmmm, that's not what my devtools are saying? Am I missing something?

```
> undefined || true
true
> undefined || false
false 
```

Reply:
Yup, indeed, you're right! Disregard my comment then.

Reply:
Nice, I learned a few things with your comments. Thank you 🫶 Done in [https://github.com/OpenRailAssociation/osrd/commit/bf7f62a68289458a4423ea06ac8a9a840110d7a1](https://github.com/OpenRailAssociation/osrd/compare/4a93d831c73b848f20ce5f8d99032647b4fe2346..bf7f62a68289458a4423ea06ac8a9a840110d7a1)